experimental/backwards-compatibility/node14/package.json

```diff
@@ -19,6 +19,6 @@
   "author": "OpenTelemetry Authors",
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=14"
+    "node": ">=18"
```

Comment:
we may want to remove that whole package (testing backwards compatibility for node 14 types)

Reply:
see https://github.com/open-telemetry/opentelemetry-js/pull/4295#discussion_r1402024336